src/Language/PureScript/Parser/Declarations.hs

```diff
@@ -399,13 +400,19 @@ parseInfixExpr
 parseHole :: TokenParser Expr
 parseHole = Hole <$> holeLit
 
-parsePropertyUpdate :: TokenParser (NonEmpty PSString, Expr)
+parsePropertyUpdate :: TokenParser [(NonEmpty PSString, Expr)]
```

Comment:
Why not just parse the tree structure directly here? It seems like now you're turning it back and forth into a list of paths.